sentry_sdk/integrations/falcon.py

```diff
@@ -186,9 +187,15 @@ def sentry_patched_prepare_middleware(
     falcon.api_helpers.prepare_middleware = sentry_patched_prepare_middleware
 
 
-def _is_falcon_http_error(ex):
-    # type: (BaseException) -> bool
-    return isinstance(ex, (falcon.HTTPError, falcon.http_status.HTTPStatus))
+def _exception_leads_to_http_5xx(ex):
+    # type: (Union[Exception, falcon.HTTPError, falcon.http_status.HTTPStatus]) -> bool
```

Comment:
I'd imagine the input (`ex`) is still of type `BaseException` -- how could we guarantee it is limited to this `Union...`? Also, note that both [`falcon.HTTPError`](https://falcon.readthedocs.io/en/stable/_modules/falcon/http_error.html) and [`falcon.http_status.HTTPStatus`](https://falcon.readthedocs.io/en/stable/_modules/falcon/http_status.html#HTTPStatus) are subtypes of `Exception`, so I guess the union is basically the same as saying `Exception`?

```suggestion
    # type: (BaseException) -> bool
```

Reply:
You are right. I will change this.